state_machine/applications/system-check/main.py

```diff
@@ -0,0 +1,94 @@
+"""
+Python system check script for PyCubed satellite board
+PyCubed Mini mainboard-v02 for Pocketqube Mission
+* Author(s): Yashika Batra
+"""
+
+from lib import pycubed
+import tests
+import tests.i2c_scan
+import tests.logging_test
+import tests.imu_test
+import tests.radio_test
+import tests.sun_sensor_test
+import tests.coil_test
+import tests.burnwire_test
+
+# initialize hardware_dict and result_dict
+hardware_dict = pycubed._cubesat.hardware
+result_dict = {
```

Comment:
What is the purpose of this?

Reply:
The hardware dictionary tells us what hardware has been initialized successfully and what hasn't. It's necessary so we can choose which tests to run and which note, but we don't have to print it (error messages for failed initializations are already printed)

The result dictionary just contains all the different tests as keys, as well as a tuple of a string and boolean as the entry (the string is the log result of the test, and the boolean is True/False for pass/fail)